src/components/Task/Task.js

```diff
@@ -1,3 +1,45 @@
-export const Task = ({ value }) => {
-  return <li>{value}</li>;
+import React from "react";
+import PropTypes from "prop-types";
+import { Button } from "../Button/Button";
+import { useRecoilState } from "recoil";
+import { ListOfTasks } from "../../store/store";
+
+export const Task = ({ value, item }) => {
+  const [tasksList, setTasksList] = useRecoilState(ListOfTasks);
+  const index = tasksList.findIndex((listItem) => listItem === item);
```

Comment:
I would recommend searching by `id` because you never know if you accidentally create a new instance somewhere, while `id` is always the real source of truth.

I also recommend moving this logic away from the component. You don't need to search for `index`. I will explain in the next comments.